src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2361 @@
+            In common_channel mode, patch embedding is independent of channels (Channel Independece). In mix_channel,
+            we follow channel independence, but in addition to patch and feature mixing, we also do channel mixing.
+            Defaults to "common_channel".
+        gated_attn (bool, *optional*): Enable Gated Attention. Defaults to False.
+        ffn (str, *optional*): MLP mode. Allowed values: mlp, gmlp. gmlp is not preferred. Defaults to "mlp".
+        self_attn (bool, *optional*): Enable Tiny self attention in addition to MLP mixing. Defaults to False.
+        self_attn_heads (bool, *optional*): Self attention heads. Defaults to 1.
+        mixer_type (str, *optional*): Mixer Type to use. Allowed values are base, gated.
+            base follows the MLP-Mixer architecture (https://arxiv.org/abs/2105.01601) gated follows the gMLP
+            architecture (https://arxiv.org/pdf/2105.08050.pdf) Defaults to "base".
+        norm_mlp (str, *optional*): Norm layer (BatchNorm or LayerNorm). Defaults to LayerNorm.
+    """
+
+    def __init__(
+        self,
+        num_patches: int,
+        patch_size: int = 16,
+        in_channels: int = 3,
+        num_features: int = 128,
+        expansion_factor: int = 2,
+        num_layers: int = 8,
+        dropout: float = 0.5,
+        mode: str = "common_channel",
+        gated_attn: bool = False,
+        ffn: str = "mlp",
+        self_attn: bool = False,
+        self_attn_heads: int = 1,
+        mixer_type: str = "base",
+        norm_mlp="LayerNorm",
+    ):
+        super().__init__()
+        self.mode = mode
+
+        self.num_patches = num_patches
+        self.patch_size = patch_size
+        self.in_channels = in_channels
+        self.num_features = num_features
+        self.num_layers = num_layers
+
+        mix_params = {}
+        if mixer_type == "base":
+            mixer_class = PatchTSMixerLayer
+        else:
+            raise Exception("mixer_type %s is not yet implemented" % (mixer_type))
+
+        self.mixers = nn.ModuleList(
+            [
+                mixer_class(
+                    num_patches=num_patches,
+                    num_features=num_features,
+                    in_channels=in_channels,
+                    expansion_factor=expansion_factor,
+                    dropout=dropout,
+                    mode=mode,
+                    gated_attn=gated_attn,
+                    ffn=ffn,
+                    self_attn=self_attn,
+                    self_attn_heads=self_attn_heads,
+                    norm_mlp=norm_mlp,
+                    **mix_params,
+                )
+                for _ in range(num_layers)
+            ]
+        )
+
+    def forward(self, x, output_hidden_states: Optional[bool] = False):
+        # flatten: [bs x num_patch x num_features]   common_channel/mix_channel: [bs x n_vars x num_patch x num_features]
+
+        all_hidden_states = []
+
+        logger.debug(x.shape)
```

Comment:
can be removed

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/46360d694e1e7f4464b268251bc8b6132db00461